eth/protocols/snap/sync.go

```diff
@@ -0,0 +1,2478 @@
+			if !delivered[res.roots[i]] {
+				res.mainTask.stateTasks[account] = res.roots[i]
+			}
+			continue
+		}
+		// State was delivered, if complete mark as not needed any more, otherwise
+		// mark the account as needing healing
+		for j, acc := range res.mainTask.res.accounts {
+			if res.roots[i] == acc.Root {
+				// If the packet contains multiple contract storage slots, all
+				// but the last are surely complete. The last contract may be
+				// chunked, so check it's continuation flag.
+				if res.subTask == nil && res.mainTask.needState[j] && (i < len(res.hashes)-1 || !res.cont) {
+					res.mainTask.needState[j] = false
+					res.mainTask.pend--
+				}
+				// If the last contract was chunked, mark it as needing healing
+				// to avoid writing it out to disk prematurely.
+				if res.subTask == nil && !res.mainTask.needHeal[j] && i == len(res.hashes)-1 && res.cont {
+					res.mainTask.needHeal[j] = true
+				}
+				// If the last contract was chunked, we need to switch to large
+				// contract handling mode
+				if res.subTask == nil && i == len(res.hashes)-1 && res.cont {
+					// If we haven't yet started a large-contract retrieval, create
+					// the subtasks for it within the main account task
+					if tasks, ok := res.mainTask.SubTasks[account]; !ok {
+						var (
+							next common.Hash
+						)
+						step := new(big.Int).Sub(
+							new(big.Int).Div(
+								new(big.Int).Exp(common.Big2, common.Big256, nil),
+								big.NewInt(storageConcurrency),
+							), common.Big1,
+						)
+						for k := 0; k < storageConcurrency; k++ {
+							last := common.BigToHash(new(big.Int).Add(next.Big(), step))
+							if k == storageConcurrency-1 {
+								// Make sure we don't overflow if the step is not a proper divisor
+								last = common.HexToHash("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
+							}
+							tasks = append(tasks, &storageTask{
+								Next: next,
+								Last: last,
+								root: acc.Root,
+							})
+							log.Debug("Created storage sync task", "account", account, "root", acc.Root, "from", next, "last", last)
+							next = common.BigToHash(new(big.Int).Add(last.Big(), common.Big1))
+						}
+						res.mainTask.SubTasks[account] = tasks
+
+						// Since we've just created the sub-tasks, this response
+						// is surely for the first one (zero origin)
+						res.subTask = tasks[0]
+					}
+				}
+				// If we're in large contract delivery mode, forward the subtask
+				if res.subTask != nil {
+					// Ensure the response doesn't overflow into the subsequent task
+					last := res.subTask.Last.Big()
+					for k, hash := range res.hashes[i] {
+						if hash.Big().Cmp(last) > 0 {
+							// Chunk overflown, cut off excess, but also update the boundary
+							for l := k; l < len(res.hashes[i]); l++ {
+								if err := res.tries[i].Prove(res.hashes[i][l][:], 0, res.overflow); err != nil {
+									panic(err) // Account range was already proven, what happened
+								}
+							}
```

Comment:
Afaict, this is the only usage of `res.tries`. This loop _looks_ like a typical "put a panic here so we can catch an errors during development". Is this loop meant to be removed eventually? 
Two reasons for asking are: 
- It obviously can panic,
- If we remove the returned trie from `VerifyRangeProof`, then we can use the stacktrie to verify the hash, 
- If we don't keep all the tries around, it decreases the overall memory requirements on the whole thing